fenix/app/src/test/java/org/mozilla/fenix/tabstray/DefaultTabsTrayControllerTest.kt

```diff
@@ -865,6 +868,42 @@ class DefaultTabsTrayControllerTest {
         verify { controller.handleNavigateToBrowser() }
     }
 
+    @Test
+    fun `GIVEN a private tab is open and selected with a normal tab also open WHEN the private tab is closed and private home page shown and normal tab is selected from tabs tray THEN normal tab is displayed  `() {
+        val normalTab = TabSessionState(
+            content = ContentState(url = "https://simulate.com", private = false),
+            id = "normalTab",
+        )
+        val privateTab = TabSessionState(
+            content = ContentState(url = "https://mozilla.com", private = true),
+            id = "privateTab",
+        )
+        browserStore = BrowserStore(
+            initialState = BrowserState(
+                tabs = listOf(normalTab, privateTab),
+                selectedTabId = privateTab.id,
+            ),
+        )
+        browsingModeManager.mode = BrowsingMode.Private
+        val controller = spyk(createController())
+
+        try {
+            mockkStatic("mozilla.components.browser.state.selector.SelectorsKt")
+            controller.handleTabSelected(privateTab, null)
+            controller.handleTabDeletion("privateTab")
+
+            verify { controller.handleTabDeletion("privateTab") }
```

Comment:
what's the goal of the `verify` here? won't it always succeed because it's being called two lines above this?

Reply:
ah I see what you mean, they don't really provide value. I'll remove the two unnecessary `verify` checks.